<!--xml-->
<launch>
    <param name="use_sim_time" value="true"/>

    <arg name="localpath" default="/home/machozhao/dataset"/>

    <node pkg="rosbag" type="play" name="rosbag" args="-r 0.2 -s 0
                $(arg localpath)/111/2.bag
                $(arg localpath)/111/3.bag  --clock ">
    </node>

 
    <node pkg="nodelet" type="nodelet" name="depth2cloud" args="standalone depth_image_proc/point_cloud_xyz">
        <remap from="/camera_info" to="/camera/depth_registered/camera_info"/>
        <remap from="/image_rect" to="/camera/depth_registered/image_raw"/>
        <remap from="points" to="/camera/depth/points2"/>
    </node>


    <!-- Random Sampling Filter : /random_sample_filter/cloud-->
    <!--node pkg="pcl_filtering" type="random_filter" name="random_filter" output="screen"
          args="input:=/camera/depth/points2" >
          <param name="sampleNum" value="4000" />
    </node-->

    <node pkg="IRONMAP_loopclosure" type="IRONmap" name="IRONmap" output="screen" args="input:=/camera/depth/points2">
        <param name="sourceFrame" type="string" value="/world"/>
        <param name="targetFrame" type="string" value="/camera_rgb_optical_frame"/>
        <param name="fixed_frame_id_" type="string" value="world"/>

        <!--depth cloud 下采样数目 -->
        <param name="DownSizeNum" type="int" value="150000"/>
        <param name="selectedPtNum" type="int" value="10000"/>
        <!-- 跳帧数 -->
        <param name="SkipframeNum" type="int" value="4"/>
        <param name="mapVoxelLeafSize" type="double" value="0.03"/>
        <param name="createGlobalMap" type="bool" value="true"/>
        <param name="loopclosure" type="bool" value="true"/>

        <!-- Show parameter in IRON-->
        <param name="subsamplingFactor" type="double" value="0.2"/>
        <param name="cellSize" type="double" value="0.075"/>
        <param name="clippingDistance" type="double" value="4.5"/>
        <param name="neighborSearchRadius" type="double" value="0.6"/>
        <param name="entropyThreshold" type="double" value="0.8"/>
        <param name="matchingTolerance" type="double" value="0.1"/>
    </node>


    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="hector_trajectory_server"
          output="screen">
        <param name="target_frame_name" type="string" value="/world"/>
        <param name="source_frame_name" type="string" value="/camera_rgb_optical_frame"/>
        <!-- <param name="source_frame_name" type="string" value="/openni_rgb_optical_frame" /> -->
        <param name="trajectory_update_rate" type="double" value="10"/>
        <param name="trajectory_publish_rate" type="double" value="10"/>
        <remap from="trajectory" to="/Trajectory"/>
    </node>

    <!-- Launch visualizaiton  -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find IRONMAP_loopclosure)/launch/directslam.rviz"/> 


    <node pkg="tf" type="static_transform_publisher" name="actualmap_to_cloudmap"
          args="0.1 0.7 0.5 0.03 3.09 1.268 /world /map 100"/>

    <!-- Show actual map in rviz-->
    <node pkg="map_server" type="map_server" name="actual_map" args="$(find IRONMAP_loopclosure)/config/mapfiles/level2.yaml"/>

</launch>

